OpenTabletDriver.Plugin/Tablet/Wheel/WheelModeManager.cs

```diff
@@ -0,0 +1,110 @@
+using System;
+using System.Collections.Generic;
+using OpenTabletDriver.Plugin.Tablet.Wheel.Actions;
+
+namespace OpenTabletDriver.Plugin.Tablet.Wheel
+{
+    public class WheelModeManager
+    {
+        private readonly WheelModeSlot[] modes;
+        private readonly Dictionary<WheelActionType, IWheelAction> actionMap;
+
+        private int currentMode = 0;
+        private bool[] previousAux = new bool[20];
+
+        private int wheelDirection = 0;
+        private DateTime lastWheelEvent = DateTime.MinValue;
+        private readonly TimeSpan repeatInterval = TimeSpan.FromMilliseconds(50);
+
+        public Action<IWheelAction, int>? OnWheelAction;
+
+        public WheelModeManager(WheelModeSlot[] config, IWheelActionFactory factory)
+        {
+            modes = config ?? throw new ArgumentNullException(nameof(config));
+
+            actionMap = new Dictionary<WheelActionType, IWheelAction>
+            {
+                { WheelActionType.Scroll, factory.CreateScroll() },
+                { WheelActionType.CanvasZoom, factory.CreateZoom() },
+                { WheelActionType.BrushSize, factory.CreateBrush() },
+                { WheelActionType.UndoRedo, factory.CreateUndoRedo() },
+                { WheelActionType.LayerUpDown, factory.CreateLayer() },
+                { WheelActionType.Custom, factory.CreateCustom() }
+            };
+        }
+
+        public WheelModeSlot CurrentMode => modes[currentMode];
+
+        public void SwitchMode()
+        {
+            int start = currentMode;
+
+            do
+            {
+                currentMode = (currentMode + 1) % modes.Length;
+            }
+            while (!modes[currentMode].Enabled && currentMode != start);
+
+            Console.WriteLine($"[Wheel] Switched to mode: {modes[currentMode].Name}");
```

Comment:
use logging api. e.g:
```suggestion
            Log.Write("WheelModeManager", $"Switched to mode: {modes[currentMode].Name}", LogLevel.Debug);
```
(syntax may be off, double check in your code)